proxmoxtf/resource/vm/vm.go

```diff
@@ -4384,13 +4406,23 @@ func vmReadCustom(
 	// Compare the initialization configuration to the one stored in the state.
 	initialization := map[string]any{}
 
-	initializationInterface := findExistingCloudInitDrive(vmConfig, vmID, "")
+	initializationInterface, initializationDevice := findExistingCloudInitDrive(vmConfig, vmID, "")
 	if initializationInterface != "" {
-		initializationDevice := getStorageDevice(vmConfig, initializationInterface)
-		fileVolumeParts := strings.Split(initializationDevice.FileVolume, ":")
+		parts := strings.Split(initializationDevice.FileVolume, ",")
+
+		fileVolume := parts[0]
+		fileVolumeParts := strings.Split(fileVolume, ":")
+
+		datastoreId := fileVolumeParts[0]
+		pathInDatastore := fileVolumeParts[1]
+		fileFormat := filepath.Ext(pathInDatastore)
 
 		initialization[mkInitializationInterface] = initializationInterface
-		initialization[mkInitializationDatastoreID] = fileVolumeParts[0]
+
+		initialization[mkInitializationDatastoreID] = datastoreId
+		if fileFormat != "" {
+			initialization[mkInitializationFileFormat] = fileFormat[1:]
+		}
```

Comment:
Does it make sense to read the device format first, and then fallback to the extension-based format if device's is not defined.  

```suggestion
		pathInDatastore := fileVolumeParts[1]

		initialization[mkInitializationInterface] = initializationInterface

		initialization[mkInitializationDatastoreID] = datastoreId
        
		if initializationDevice.Format != nil && *initializationDevice.Format != "" {
			initialization[mkInitializationFileFormat] = *initializationDevice.Format
		} else {
			fileFormat := filepath.Ext(pathInDatastore)
			if fileFormat != "" {
				initialization[mkInitializationFileFormat] = fileFormat[1:]
			}
		}
```